frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx

```diff
@@ -241,11 +241,20 @@ export default class TagEditorParam extends Component {
           <div className="pb3">
             <h4 className="text-medium pb1">{t`Default filter widget value`}</h4>
             <ParameterValueWidget
-              parameter={{
-                type:
-                  tag["widget-type"] ||
-                  (tag.type === "date" ? "date/single" : null),
-              }}
+              parameter={
+                tag.type === "dimension"
+                  ? parameter || {
+                      ...tag,
+                      type:
+                        tag["widget-type"] ||
+                        (tag.type === "date" ? "date/single" : null),
+                    }
+                  : {
+                      type:
+                        tag["widget-type"] ||
+                        (tag.type === "date" ? "date/single" : null),
+                    }
+              }
```

Comment:
is a `tag.type` of `"dimension"` sufficient to mean we are intending to filter on a specific field or are there other things I need to concern myself with here?

Not really clear on the whole `tag` thing but as it stands in order to make the new parameter types work we need to pass in the real `parameter` object vs this fake `{type: ...}` object we were passing in before -- but we still need the `{type}` object to render `TextWidget` in scenarios where we're creating a text filter.

Reply:
pretty sure `dimension` always means a Field

Reply:
Tags are anything inside `{{...}}` in a native query. They can be "normal" variables (text, number, date, or field filters), snippets, or cards. Field filters' type is "dimension" and they always refer to a single field.